First pull request

MarinaBerezovikova opened the pull request:
**1 - delete repeating input a date in the wrong input case**
Description: In the input process was appearing bug by verifying inputted data. Can suppose we will fix that issue in the future version.
**2 - do read and write resources from resources folder**
Description: It's opening the possible to start application from any another pc.
**3 - delete ResourseManager class**
Description: It was deleted as unused.
**4 - add JsonFileReader and JsonFileWriter**
Description: We were created two classes to realize read and write a JSON file. I separated functional to two classes for better code reading and more comfortable extension code.

MarinaBerezovikova committed:
- delete repeating input a date in the wrong input case
- do read and write resources from resources folder
- delete ResourseManager class
- add JsonFileReader and JsonFileWriter

ivanChernik commented on pom.xml:
```diff
@@ -19,6 +19,11 @@
+        <dependency>
+            <groupId>commons-io</groupId>
+            <artifactId>commons-io</artifactId>
+            <version>2.13.0</version>
```
extract versions to properties like tyijs
![image](https://github.com/MarinaBerezovikova/notebook-console/assets/9913515/043a775f-c936-4245-a4fd-0b3311933fe4)

ivanChernik commented on src/main/java/com/company/notebook/Driver.java:
```diff
@@ -4,16 +4,18 @@
 public class Driver {
 
-    static String path = "e:\\LocalRepositoryGit\\data_note_6task.json";
+    static String fileName = "data_note.json";
```
private FILE_NAME

ivanChernik commented on src/main/java/com/company/notebook/JsonFileWriter.java:
```diff
@@ -0,0 +1,23 @@
+public class JsonFileWriter {
+
+    Gson gson = new GsonBuilder().setPrettyPrinting().create();
+    String filePath = "src/main/resources/data_note.json";
```
filename should be in one place

ivanChernik commented:
Notebook to domain package

ivanChernik commented on src/main/java/com/company/notebook/JsonFileReader.java:
```diff
@@ -0,0 +1,58 @@
+    Gson gson = new GsonBuilder().setPrettyPrinting().create();
+    List<Note> listNotesFromGson = new ArrayList<>();
+
+    public List<Note> readResource() throws IOException {
```
filename as parameter

ivanChernik replied:
trows to try catch block

ivanChernik commented on src/main/java/com/company/notebook/JsonFileReader.java:
```diff
@@ -0,0 +1,58 @@
+        }
+    }
+
+    private List<Note> setContentToList(InputStream is) {
```
is to inputStream

ivanChernik replied:
parseNoteList

ivanChernik commented on src/main/java/com/company/notebook/JsonFileReader.java:
```diff
@@ -0,0 +1,58 @@
+public class JsonFileReader {
+
+    String fileName = Driver.fileName;
+    Gson gson = new GsonBuilder().setPrettyPrinting().create();
```
private

ivanChernik commented on src/main/java/com/company/notebook/JsonFileReader.java:
```diff
@@ -0,0 +1,58 @@
+
+    String fileName = Driver.fileName;
+    Gson gson = new GsonBuilder().setPrettyPrinting().create();
+    List<Note> listNotesFromGson = new ArrayList<>();
```
return in method as new local var

ivanChernik commented on src/main/java/com/company/notebook/JsonFileReader.java:
```diff
@@ -0,0 +1,58 @@
+            BufferedReader reader = new BufferedReader(new InputStreamReader(is, StandardCharsets.UTF_8));
+
+            Type listType = new TypeToken<ArrayList<Note>>() {}.getType();
+            listNotesFromGson = gson.fromJson(reader, listType);
```
return listNotesFromGson

ivanChernik commented on src/main/java/com/company/notebook/JsonFileReader.java:
```diff
@@ -0,0 +1,58 @@
+            listNotesFromGson = gson.fromJson(reader, listType);
+            reader.close();
+        } catch (Exception e) {
+            System.out.println("In process appeared some problem. A specific description is below.");
```
create new exception and throw it here

ivanChernik replied:
extebd RuntimeException

ivanChernik commented on src/main/java/com/company/notebook/JsonFileWriter.java:
```diff
@@ -0,0 +1,23 @@
+    Gson gson = new GsonBuilder().setPrettyPrinting().create();
+    String filePath = "src/main/resources/data_note.json";
+
+    public void fileWriter() {
```
paramenter filePath and notes

ivanChernik commented on src/main/java/com/company/notebook/JsonFileWriter.java:
```diff
@@ -0,0 +1,23 @@
+            gson.toJson(Notebook.notes, writer);
+            writer.flush();
+        } catch (IOException e) {
+            e.printStackTrace();
```
create and throw custom exception

ivanChernik replied:
https://www.tutorialspoint.com/how-to-create-a-custom-unchecked-exception-in-java

MarinaBerezovikova committed:
modificate pom.xml file Resolves #discussion_r128703002

MarinaBerezovikova committed:
modificate pom.xml file Resolves #discussion_r128703002

MarinaBerezovikova committed:
add packages of project
add custom Exceptions ParseNoteListException FileWriterException
refactor JsonFileManager class
do all changes from pull request comments
